### PR TITLE
Add post-create hook

### PR DESCRIPTION
**Pull request by josegonzalez**
Note that this is invoked *before* the application is deployed, so it's usage is quite different than a potential post-successful-create hook.

### CODE REVIEW

File: plugins/apps/commands [hunk 1]
```diff
@@ -16,6 +16,7 @@ case "$1" in
 
     mkdir -p "$DOKKU_ROOT/$APP"
     echo "Creating $APP... done"
+    plugn trigger post-create $APP
```
Comment (michaelshobbs) [inline plugins/apps/commands]: `post-app-create` perhaps?

File: plugins/git/commands [hunk 2]
```diff
@@ -14,6 +14,7 @@ git_build_app_repo() {
   chmod 755 $TMP_WORK_DIR
   unset GIT_DIR GIT_WORK_TREE
   pushd $TMP_WORK_DIR > /dev/null
+  [[ ! -d "$DOKKU_ROOT/$APP" ]] && dokku apps:create $APP
```
Comment (michaelshobbs) [inline plugins/git/commands]: Why is this necessary?

Comment (josegonzalez) [inline plugins/git/commands]: Otherwise we won't actually call the hook on every app creation. If I push a repo up without first creating the app, it skips the above...

Comment (josegonzalez) [inline plugins/git/commands]: We could also just manually call the hook here instead.

File: plugins/apps/commands (hunk 1 above)
Comment (josegonzalez) [inline plugins/apps/commands]: The analogue for app:destroy is `post-delete`.

Comment (michaelshobbs) [inline plugins/apps/commands]: :+1:

File: plugins/git/commands (hunk 2 above)
Comment (michaelshobbs) [inline plugins/git/commands]: I'd personally prefer this to be a function call rather than an invocation of a new `dokku` process.
